pkg/cfn/builder/nodegroup.go

```diff
@@ -114,28 +114,18 @@ func (n *NodeGroupResourceSet) addResourcesForNodeGroup() error {
 	}
 
 	if volumeSize := n.spec.VolumeSize; volumeSize != nil && *volumeSize > 0 {
+		launchTemplateData.BlockDeviceMappings = []gfn.AWSEC2LaunchTemplate_BlockDeviceMapping{{
+			DeviceName: gfn.NewString(*n.spec.VolumeName),
+			Ebs: &gfn.AWSEC2LaunchTemplate_Ebs{
+				VolumeSize: gfn.NewInteger(*volumeSize),
+				VolumeType: gfn.NewString(*n.spec.VolumeType),
+				Encrypted: gfn.NewBoolean(*n.spec.VolumeEncrypted),
+				KmsKeyId: gfn.NewString(*n.spec.VolumeKmsKeyID),
```

Comment:
I think you forgot to remove the kmskeyid from the original object

Reply:
Ah, thanks! I have updated.